server/pulp/server/webservices/views/util.py

```diff
@@ -46,3 +49,22 @@ def generate_redirect_response(response, href):
     response.status_code = 201
     response.reason_phrase = 'CREATED'
     return response
+
+
+def json_body_required(func, allow_empty=False):
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        request = args[1]
+        if allow_empty and not request.body:
+            request.body_as_json = {}
+            return func(*args, **kwargs)
+
+        try:
+            request.body_as_json = json.loads(request.body)
+        except ValueError:
+            raise PulpCodedValidationException(error_code=error_codes.PLP1009)
```

Comment:
It's strange to me that the coded validation exceptions aren't raised as: `raise PulpException1009`